Header header
int32 goal
int32 cmd_type